dotfiles/.install_on_demand/.etcdctl

```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env bash
+
+alias etcdctl="etcdctl_install"
+function etcdctl_install() {
+  yes="Yy"
+  no="Nn"
+  app="etcdctl"
+  install_func="_install_etcdctl"
+
+
+  if ! hash "${app}" &> /dev/null; then
+    local _confirmation=""
+    while [[ ! $_confirmation =~ ^[$yes$no]$ ]]; do
+          read -p "${app} not found. Do you want to install it now? (Y/n) " -r _confirmation
+          if [[ $_confirmation =~ ^[$no]$ ]]; then
```

Comment:
```bash
                     _                            _          _ _
  __ _  __ _ _ __ __| | ___ _ __   ___ _ __   ___| |__   ___| | |
 / _` |/ _` | '__/ _` |/ _ \ '_ \ / _ \ '__| / __| '_ \ / _ \ | |
| (_| | (_| | | | (_| |  __/ | | |  __/ |    \__ \ | | |  __/ | |
 \__, |\__,_|_|  \__,_|\___|_| |_|\___|_|    |___/_| |_|\___|_|_|
 |___/

Run ghelp to get information about installed tools and packages

root at 3438ec5ec3ad in /
$ etcdctl
etcdctl not found. Do you want to install it now? (Y/n)
etcdctl not found. Do you want to install it now? (Y/n)
etcdctl not found. Do you want to install it now? (Y/n)
```

Y is the default (as indicated by its uppercase), but hitting enter will not select the default option, which is ‘yes’.

Reply:
I copied this from `wg`.
Will adapt accordingly.